relay-general/src/store/transactions/processor.rs

```diff
@@ -396,6 +414,53 @@ impl Processor for TransactionsProcessor<'_> {
         Ok(())
     }
 
+    /// Validates the length and characters of measurement names.
+    fn process_measurements(
+        &mut self,
+        value: &mut crate::protocol::Measurements,
+        meta: &mut Meta,
+        _state: &ProcessingState<'_>,
+    ) -> ProcessingResult {
+        let measurements = &mut value.0;
+        let mut removed_measurements = Object::new();
+
+        measurements.retain(|key, value| {
+            let measurement = match value.value_mut() {
+                Some(m) => m,
+                None => return false,
+            };
+
+            if !is_valid_metric_name(key) {
+                removed_measurements
+                    .insert(key.clone(), Annotated::new(std::mem::take(measurement)));
+                return false;
+            }
+
+            let name_len = key.len();
+            let unit_len = measurement.unit.value().map(|val| val.to_string().len());
+
+            if let (Some(fixed_len), Some(max_len), Some(unit_len)) = (
+                self.measurement_config.fixed_length,
+                self.measurement_config.max_len,
+                unit_len,
+            ) {
+                if (name_len + unit_len + fixed_len) > max_len {
```

Comment:
There's no need to keep adding the `fixed_len` in every evaluation, better to move it to the right side of the equation:

```suggestion
                if (name_len + unit_len) > name_unit_max_len {
```

Reply:
makes sense, I've now added `max_name_and_unit_len` to `lightnormalizationconfig` so it gets calculated a lot earlier.